Source/Dafny/AST/DafnyAst.cs

```diff
@@ -5573,7 +5573,7 @@ public NewtypeDecl(IToken tok, string name, ModuleDefinition module, BoundVar bv
     FreshIdGenerator RedirectingTypeDecl.IdGenerator { get { return IdGenerator; } }
 
     bool ICodeContext.IsGhost {
-      get { throw new NotSupportedException(); }  // if .IsGhost is needed, the object should always be wrapped in an CodeContextWrapper
+      get { return false; }
```

Comment:
@robin-aws  Do you know why it said "the object should always be wrapped in a codecontextwrapper"? In my case, the answer to IsGhost should clearly be false.

Reply:
The reason is that `newtype`s and subset types essentially have _two_ contexts. One context is the constraint, which is ghost. The other context is the witness, which may or may not be ghost. Therefore, `ICodeContext.IsGhost` for `NewtypeDecl` and`TypeSynonymDeclBase` cannot be determined. Instead, this type declaration is first placed in a `CodeContextWrapper`, which keeps track of whether or not it's a ghost context (search for occurrences of `new CodeContextWrapper` in `Resolver.cs`). By throwing an exception here, we catch places in the code that forgot to do the wrapping.

Reply:
In the newest version, I reverted those changes, since I'm using a different way to test ghostness